plugin/src/main/java/org/opensearch/sql/plugin/rest/RestPPLQueryAction.java

```diff
@@ -134,6 +135,11 @@ public void onFailure(Exception e) {
                       "Failed to explain the query due to error: " + e.getMessage());
                 } else if (e instanceof IllegalAccessException) {
                   reportError(channel, e, BAD_REQUEST);
+                } else if (e instanceof PrometheusClientException) {
```

Comment:
why PrometheusClientException special is handled in here. Does line 154 cover RuntimeException already?

Reply:
AppSec recommends to send 400 incase of Prometheus Exceptions because only customer can fix those issues.
Line154 sends 500 error which is not recommended.

Reply:
So this change is same as adding `PrometheusClientException` to `isClientError`?

Reply:
Changes are already in progress.